### PR TITLE
Registered azure openai component

### PR DESCRIPTION
**Pull request by shivam-51**
# Description

Add azure openai component to dapr. Refer to this PR for more details https://github.com/dapr/components-contrib/pull/2965

## Issue reference

https://github.com/dapr/components-contrib/issues/2716

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by shivam-51** [comment]
@ItalyPaleAle Please have a look.

File: cmd/daprd/components/binding_azure_openai.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//go:build allcomponents || stablecomponents
```
Comment (ItalyPaleAle) [inline cmd/daprd/components/binding_azure_openai.go:1]:
```suggestion
//go:build allcomponents
```

This component is not stable :)

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Thanks!

Please also run:

```
go get -u github.com/dapr/components-contrib@master
make modtidy-all
```

**Review by ItalyPaleAle — APPROVED** [review]

**Review by ItalyPaleAle — DISMISSED** [review]

**Comment by ItalyPaleAle** [comment]
@mukundansundar docs are here: https://github.com/dapr/docs/pull/3618

@shivam-51 looks like there's a failure building this, could you please take a look?

**Comment by mukundansundar** [comment]
> @mukundansundar docs are here: [dapr/docs#3618](https://github.com/dapr/docs/pull/3618)
> 
> @shivam-51 looks like there's a failure building this, could you please take a look?

interesting that the label did not work and automatically create a docs issue ... will have a look at the workflow for it.

**Comment by shivam-51** [comment]
@ItalyPaleAle Please have a look. The linting should not fail now.

**Pull request merged**